diff --git a/Classes/Command/SetupCommand.php b/Classes/Command/SetupCommand.php
index a52afe5c1fb..aaec25928d3 100644
--- a/Classes/Command/SetupCommand.php
+++ b/Classes/Command/SetupCommand.php
@@ -432,7 +432,6 @@ EOT
                     $default = $this->getDefinition()->getOption($key)->getDefault();
                     $defaultLabel = empty($value) ? '' : ' [default: ' . $default . ']';
                     $question = new Question('Enter the database "' . $key . '"' . $defaultLabel . ' ? ', $default);
-
                     if ($key === 'password') {
                         $question = new Question('Enter the database "' . $key . '" ? ', $default);
                         $question->setHidden(true);
@@ -445,19 +444,18 @@ EOT
                                     1669747572
                                 );
                             }
-
                             return $host;
                         };
                         $question->setValidator($hostValidator);
                     } elseif ($key === 'port') {
                         $portValidator = function ($port) {
-                            if (!$this->setupDatabaseService->isValidDbPort((int)$port)) {
+                            $port = (int)$port;
+                            if (!$this->setupDatabaseService->isValidDbPort($port)) {
                                 throw new \RuntimeException(
                                     'Please use a port in the range between 1 and 65535.',
                                     1669747592,
                                 );
                             }
-
                             return $port;
                         };
                         $question->setValidator($portValidator);
@@ -469,12 +467,10 @@ EOT
                                     1669747601,
                                 );
                             }
-
                             return $value;
                         };
                         $question->setValidator($emptyValidator);
                     }
-
                     if ($envValue === false && $key === 'password') {
                         // Force this question if no `TYPO3_DB_PASSWORD` set via cli.
                         // Thus, the user will always be prompted for a password even --no-interaction is set.
@@ -491,7 +487,6 @@ EOT
                         $envValue = $envValue ?: $default;
                         $value = $validator ? $validator($envValue) : $envValue;
                     }
-
                     $databaseConnectionOptions[$key] = $value;
             }
         }
diff --git a/Classes/Service/SetupDatabaseService.php b/Classes/Service/SetupDatabaseService.php
index 11714d955ce..85ebd65ffff 100644
--- a/Classes/Service/SetupDatabaseService.php
+++ b/Classes/Service/SetupDatabaseService.php
@@ -597,7 +597,7 @@ class SetupDatabaseService
                 'driver' => 'mysqli',
                 'username' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['user'] ?? '',
                 'password' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['password'] ?? '',
-                'port' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 3306,
+                'port' => (int)($GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 3306),
                 'database' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['dbname'] ?? '',
             ];
             $host = $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['host'] ?? '127.0.0.1';
@@ -629,7 +629,7 @@ class SetupDatabaseService
                 'driver' => 'pdo_mysql',
                 'username' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['user'] ?? '',
                 'password' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['password'] ?? '',
-                'port' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 3306,
+                'port' => (int)($GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 3306),
                 'database' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['dbname'] ?? '',
             ];
             $host = $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['host'] ?? '127.0.0.1';
@@ -666,7 +666,7 @@ class SetupDatabaseService
                 'username' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['user'] ?? '',
                 'password' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['password'] ?? '',
                 'host' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['host'] ?? '127.0.0.1',
-                'port' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 5432,
+                'port' => (int)($GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['port'] ?? 5432),
                 'database' => $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['dbname'] ?? '',
             ];
             if (($GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][ConnectionPool::DEFAULT_CONNECTION_NAME]['driver'] ?? '') === 'pdo_pgsql') {
